### PR TITLE
Removing olm dependency

### PR DESCRIPTION
**Pull request by Bowenislandsong**
This PR follows removing dependency olm pr https://github.com/operator-framework/api/pull/14 from api repo. With api adding necessary registry dependency of olm, registry will only need to rely on api repo and remove the cyclic dependecy. Source of truth will be on api repo.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by Bowenislandsong** [comment]
waiting on https://github.com/operator-framework/api/pull/14 to merge first

File: go.mod [hunk 1]
```diff
@@ -15,7 +15,6 @@ require (
 	github.com/onsi/ginkgo v1.10.1
 	github.com/onsi/gomega v1.7.0
 	github.com/operator-framework/api v0.0.0-20200120235816-80fd2f1a09c9
```
Comment (ecordell) [inline go.mod]:
```suggestion
	github.com/operator-framework/api v0.1.0
```

**Pull request merged**